### ISSUE TITLE
[generate:plugin] Use ModifiedResourceResponse instead of ResourceResponse

### ISSUE THREAD

**Issue by codesmithsf** [hechoendrupal/drupal-console#3266]
For the [generate:plugin:rest:resource] command, the templated code has this line at the end of each of HTTP state/verb functions:

`return new ResourceResponse("Implement REST State HTTP_STATE!");`

I believe ResourceResponse should be ModifiedResourceResponse for the POST, PATCH and DELETE states. See

https://api.drupal.org/api/drupal/core!modules!rest!src!ModifiedResourceResponse.php/8.3.x

**Comment by codesmithsf** [comment]
And actually DELETE should be this

`return new ModifiedResourceResponse(NULL, 204);`

since no Body content is returned.

**Comment by jmolivas** [comment]
@codesmithsf are you planning to send a PR to have this fixed?

**Comment by codesmithsf** [comment]
Sure, I can do that. Will get to it in the next day or two. Thanks.

**Comment by LOBsTerr** [comment]
PR https://github.com/hechoendrupal/drupal-console/pull/3666

1) I have changed the code of the twig template to generate more universal methods based on comments above;
2) I moved the HTTP methods to separate method and add settings per method (HTTP code, HTTP Response class). So, we can use them in the twig template later
3) I noticed that we don't have any validator for HTTP methods. Basically we can pass any string when use options. Now, the code will check the passed HTTP methods based on the method created in the 2 point
